src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -119,47 +122,50 @@ class RegularSync(
   def handleNewBlockMessages: Receive = {
     case MessageFromPeer(NewBlock(newBlock, _), peerId) =>
       //we allow inclusion of new block only if we are not syncing
-      if (notDownloading() && topOfTheChain) {
+      if (notDownloading() && topOfTheChain && !importingBlocks) {
+        importingBlocks = true
```

Comment:
not sure what you mean , but this flag is necessary  to prevent importing two blocks at the same time, second execution context would not help much here. Although i have in mind moving all block importing activities to separate execution context (instead of default one) in next pr-s

Reply:
hmm there is already pr to refactor regular sync , so in this one i would stick to this boolean variable, as i have it already pretty well tested ( i am keeping mantis on top etc network for around a week without problems)